docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,173 @@
+```
+
+The configuration source and the corresponding format you should use is use-case specific. It depends on the platform the server is deployed to
+as well as on the optimizations you can get for an optimal runtime.
+
+== Configuration Stages
+
+The server options are narrowed to a specific command or configuration stage. This clear separation is one of the key aspects
+of the server configuration and is related to a series of optimizations that are done in order to deliver the best runtime when starting and running the server.
+
+Configuration is basically done in two stages:
+
+* Before starting the server in order to build an optimized server image for an optimal runtime
+* When starting the server
+
+The first stage involves running the `build` command using any **build option** available from this command. On the other hand, the second stage involves
+starting the server using any **configuration option** available from the `start` command.
+
+== Configuring the server
+
+The clear separation between build options and configuration options is one of the key aspects
+of the server configuration and related to a series of optimizations that are done in order to deliver the best runtime when starting and running the server.
+
+The first step when configuring the server is to use the `build` command to set any build option you want to change. For instance, to choose a specific database vendor.
+By looking at the individual configuration options at <@links.server id="all-config"/>, you notice
+that some options are marked with an icon to indicate they are a build option. In other words, they can only be set and only take effect when running the `build` command.
+You can also check all the options that require a build by looking at the `help` message of the `build` command:
+
+<@kc.build parameters="--help"/>
+
+The `build` command is responsible for producing an immutable and optimized server image (similar concept to building a container image). In addition to persisting
+any build option you have set, this command also performs a series of optimizations to deliver the best runtime when starting and running the server. Basically,
+a lot of processing that would usually happen when starting and running the server is no longer necessary and the server can start and run faster.
+
+Some optimizations performed by the `build` command are:
+
+* Closed-world assumption about installed providers, hence no need to re-create the registry every time the server starts
+* Configuration files are pre-parsed to reduce IO when starting the server
+* Database specific resources are configured and prepared to run against a specific database vendor
+* By persisting build options into the server image, the server does not perform any additional step to interpret configuration options and (re)configure itself
+
+Once you run the `build` command, you won't need to set the same build options again (they are now persisted) so that you can just `start` the server:
```

Comment:
Change
again (they are now persisted) so that you can just `start` the server:
TO
again, because they are now persisted.  You can just `start` the server: